cpp/oneapi/dal/algo/covariance/parameters/cpu/compute_parameters.cpp

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <algorithm>
+
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/detail/profiler.hpp"
+
+#include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+
+#include "oneapi/dal/algo/covariance/common.hpp"
+#include "oneapi/dal/algo/covariance/compute_types.hpp"
+
+#include "oneapi/dal/algo/covariance/parameters/cpu/compute_parameters.hpp"
+
+namespace oneapi::dal::covariance::parameters {
+
+using dal::backend::context_cpu;
+
+template <typename Float>
+std::int64_t propose_block_size(const context_cpu& ctx, const std::int64_t row_count) {
+    std::int64_t block_size = 140l;
+    if (ctx.get_enabled_cpu_extensions() == dal::detail::cpu_extension::avx512) {
+        if (5000l < row_count && row_count <= 50000l) {
+            block_size = 1024l;
+        }
+    }
+    return block_size;
```

Comment:
Return values should be tested in a unit test.

Reply:
There is no much sense in testing those values. Because they are hard-coded only for now. Further this logic will be replaced by some ML model that will compute those return values on the fly.
So, only the ranges of those values, but not the exact values, can be checked.
I will add the ranges test.

Reply:
I disagree, because unit tests also serve as a MWE of using a function, and as such they help the code to be self-documenting.
But you are right in the sense that the benefit is limited, so don't worry about it if it's high effort. Thank you for the explanation!